SMTP notify enhancements: full HTML emails and custom `product_name` in email headers

azogue opened the pull request:
## Description:

- Ability to send full HTML emails, with or without inline attached images.
- Customize `product_name` identifier in the `FROM` and `X-MAILER` email headers.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2607

## Example entry for `configuration.yaml` (if applicable):
```yaml
notify: 
- name: NOTIFIER_NAME
   platform: smtp
  server: smtp.gmail.com 
  port: 587
  timeout: 15 
  sender: john@gmail.com 
  starttls: true 
  username: john@gmail.com 
  password: thePassword 
  recipient:
       - james@gmail.com
       - bob@gmail.com
  product_name: My Home
```

## Checklist:
  - [X] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)
  - [X] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

azogue committed:
SMTP notify enhancements: HTML emails and customization options

- Send full HTML emails, with or without inline attached images.
- Custom `timeout`.
- Custom `product_name` identifier for the `FROM` and `X-MAILER` email headers.
- New HTML email test

pvizeli commented:
What is the benefit for custom `X-MAILER`? I see only a grow up config

azogue commented:
The `X-MAILER` header is changed for consistency with the change in the `FROM` header, it does not have its specific option in the configuration.

The change in the `FROM` header does seem to me relevant as it **sets the display name on the e-mail readers**, instead of the ugly raw e-mail. More so **when handling multiple instances** of Home Assistant.

balloob commented on homeassistant/components/notify/smtp.py:
```diff
@@ -128,20 +135,26 @@ def send_message(self, message="", **kwargs):
-        msg['From'] = self._sender
-        msg['X-Mailer'] = 'HomeAssistant'
+        msg['From'] = '{} <{}>'.format(self._product_name, self._sender)
+        msg['X-Mailer'] = self._product_name
```
The X-Mailer should stick to `HomeAssistant`, as that's the product that emails it.

balloob commented on homeassistant/components/notify/smtp.py:
```diff
@@ -47,6 +50,7 @@
     vol.Optional(CONF_STARTTLS, default=DEFAULT_STARTTLS): cv.boolean,
     vol.Optional(CONF_USERNAME): cv.string,
     vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_PRODUCT_NAME, default=DEFAULT_PRODUCT_NAME): cv.string,
```
I don't like the word product name, can you use CONF_SENDER_NAME instead?

azogue committed:
`sender_name` instead of product_name

 - Change `sender_name` instead of `product_name`.
 - No changes in `X-Mailer` header.
 - `From` header as always unless you define the new `sender_name` parameter.

azogue commented:
@balloob, changes are done here and in the doc, so I think it's ready.

Do you want me to edit the PR subject and description?

balloob reviewed (approved):
Awesome, thanks 🐬
